### PR TITLE
[SPIRV] Add support of [[vk::ext_type_def]]

### PR DESCRIPTION
**Pull request by jiaolu**
this is related
https://github.com/microsoft/DirectXShaderCompiler/issues/3919

### CODE REVIEW

**Comment by jaebaek** [comment]
I am really sorry for being late. I am experiencing intensive workloads. I will review it until tomorrow. I know this is a complicated one and I appreciate your strong contribution!

**Review by jaebaek — COMMENTED** [review]
I will check how to define `vk::ext_type`, but could you please address my comments?
Let me know if you think sth of them are unclear (We can have email or video-call communications as well).

**Comment by jaebaek** [comment]
@jiaolu I sent a commit to your branch that allows us to put `ext_type` into `vk` namespace.

**Comment by jiaolu** [comment]
> @jiaolu I sent a commit to your branch that allows us to put `ext_type` into `vk` namespace.



> @jiaolu I sent a commit to your branch that allows us to put `ext_type` into `vk` namespace.

I made stupid mistake of overriding your vk namespace commit when i rebased the PR. 
Sorry .

**Comment by jiaolu** [comment]
hi, @jaebaek, 
 any comments to the latest change to the commits?

File: tools/clang/lib/SPIRV/SpirvContext.cpp [hunk 1]
```diff
@@ -527,5 +527,23 @@ void SpirvContext::moveDebugTypesToModule(SpirvModule *module) {
   typeTemplateParams.clear();
 }
 
+const SpirvIntrinsicType *SpirvContext::getSpirvIntrinsicType(
+    unsigned typeId, unsigned typeOpCode,
+    llvm::ArrayRef<SpvIntrinsicTypeOperand> operands) {
+  if (spirvIntrinsicTypes[typeId] == nullptr) {
+    spirvIntrinsicTypes[typeId] =
+        new (this) SpirvIntrinsicType(typeOpCode, operands);
```
Comment (jaebaek) [inline tools/clang/lib/SPIRV/SpirvContext.cpp:535]: We have to free this allocation in the destructor of `SpirvContext`.
In `SpirvContext::~SpirvContext()`, please add
```
for (auto itr: spirvIntrinsicTypes)
  if (itr->second)
   delete itr->second;
```

**Review by jaebaek — COMMENTED** [review]
Sorry for being late.
LGTM. I have just a last comment.

Thanks!

**Review by jaebaek — APPROVED** [review]

**Pull request merged**